app/src/models/repository.ts

```diff
@@ -15,23 +15,37 @@ function getBaseName(path: string): string {
   return baseName
 }
 
+/** Base type for a directory you can run git commands successfully */
+export type WorkingTree = {
+  readonly path: string
+}
+
 /** A local repository. */
 export class Repository {
   public readonly name: string
+  /** The main working tree's path
+   * (also what we commonly think of as the repository's working directory)
+   */
+  private readonly mainWorkTree: WorkingTree
```

Comment:
@outofambit can you tell me a bit about how you envision this being used? It's `private` at the moment, and it'd be cool to think about what should be accessing this and why it might need to?

Reply:
honestly i'm not sure yet, which is why its marked `private` for now. mostly, i'd like our model to reflect the relationships between Git concepts, as much for accuracy as for making these things easier to understand for people coming into the codebase.

my first thought was to replace the use of `Repository.path` with `Repository.mainWorkTree.path` but that would be a much larger (though not complicated) refactor and wouldn't get us immediate material benefits. so having that internally with the same `.path` api seemed like a decent compromise to start from.

Reply:
I'm a bit hesitant to move `path` under `mainWorkTree` and make callers aware of that, so what you've done here makes sense. Just some food for thought as you're making changes to these internals.